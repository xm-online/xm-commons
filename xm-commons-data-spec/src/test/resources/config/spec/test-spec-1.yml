---
test-items:
    - key: PRODUCT
      inputSpec: |-
          {
            "type": "object",
            "properties": {},
            "$ref": "#/xmDefinition/productData",
            "orderedBy": { "$ref": "#/xmDefinition/employeeInfo" }
          }
      inputForm: |-
          {
            "form": [
              {
                "$ref": "#/xmForm/productForm",
                "employee": {
                  "items": [
                    { "index": 1 },
                    { "$ref": "#/xmForm/singleArrayForm", "key": "userKey[]" },
                    { "index": 2 }
                  ]
                }
              },
              { "$ref": "#/xmForm/formWithSubForm", "key": "prefix1" },
              { "$ref": "#/xmForm/formWithSubFormWithKey", "key": "prefix2" },
              { "lastKey": "lastValue" }
            ]
          }
