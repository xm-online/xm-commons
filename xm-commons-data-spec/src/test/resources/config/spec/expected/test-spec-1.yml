test-items:
    - key: PRODUCT
      inputSpec: |
          {
            "type": "object",
            "properties": {},
            "$ref": "#/xmDefinition/productData",
            "orderedBy": {
              "$ref": "#/xmDefinition/employeeInfo"
            },
            "xmDefinition": {
              "productData": {
                "$schema": "http://json-schema.org/draft-07/schema#",
                "type": "object",
                "properties": {
                  "productId": {
                    "type": "number"
                  },
                  "startDate": {
                    "type": "string"
                  },
                  "endDate": {
                    "type": "string"
                  },
                  "budget": {
                    "type": "number",
                    "minimum": 0
                  },
                  "paymentMode": {
                    "type": "string",
                    "enum": [
                      "online",
                      "offline"
                    ],
                    "default": "online"
                  },
                  "balances": {
                    "$ref": "#/xmDefinition/balances"
                  }
                },
                "required": [
                  "productId",
                  "budget",
                  "paymentMode"
                ]
              },
              "balances": {
                "type": "object",
                "properties": {
                  "budgetBalance": {
                    "type": "number"
                  }
                }
              },
              "employeeInfo": {
                "$schema": "http://json-schema.org/draft-07/schema#",
                "type": "object",
                "properties": {
                  "storeName": {
                    "type": "string"
                  },
                  "info": {
                    "$ref": "#/xmDefinition/userInfo"
                  }
                }
              },
              "userInfo": {
                "$schema": "http://json-schema.org/draft-07/schema#",
                "type": "object",
                "properties": {
                  "username": {
                    "type": "string"
                  },
                  "phoneNumber": {
                    "type": "string"
                  },
                  "age": {
                    "type": "number"
                  }
                }
              }
            }
          }
      inputForm: |
          {
            "form": [
              {
                "employee": {
                  "items": [
                    {
                      "index": 1
                    },
                    {
                      "name": "testName",
                      "key": "userKey[].fieldName"
                    },
                    {
                      "index": 2
                    }
                  ]
                },
                "form": [
                  {
                    "key": "budget",
                    "title": {
                      "en": "Budget"
                    },
                    "type": "number"
                  },
                  {
                    "type": "flex",
                    "flex-flow": "row wrap",
                    "fxLayoutAlign": "space-between center",
                    "fxLayout": "row wrap",
                    "justify-content": "space-between",
                    "items": [
                      {
                        "key": "codeExpirationTime",
                        "title": {
                          "en": "Verification Code Expiration Time",
                          "uk": "Час дії коду підтвердження"
                        },
                        "type": "datetime-picker",
                        "formatString": "DD-MM-YYYY HH:mm",
                        "readonly": true,
                        "flex": "0 0 32%"
                      },
                      {
                        "key": "acceptanceTimeOfPrivacyPolicy",
                        "title": {
                          "en": "Private policy accepted at",
                          "uk": "Політика конфіденційності прийнята о"
                        },
                        "type": "datetime-picker",
                        "formatString": "DD-MM-YYYY HH:mm",
                        "readonly": true,
                        "flex": "0 0 32%"
                      },
                      {
                        "key": "acceptanceTimeOfContactDetailProcessing",
                        "title": {
                          "en": "Processing of contact details agreed at",
                          "uk": "Обробка контактних даних погоджена о"
                        },
                        "type": "datetime-picker",
                        "formatString": "DD-MM-YYYY HH:mm",
                        "readonly": true,
                        "flex": "0 0 32%"
                      }
                    ]
                  },
                  {
                    "key": "paymentMode",
                    "title": {
                      "en": "Payment mode"
                    },
                    "type": "string"
                  }
                ]
              },
              {
                "key": "prefix1.formWithSubFormField",
                "type": "string"
              },
              {
                "key": "prefix1.subForm",
                "type": "string"
              },
              {
                "key": "prefix2.formWithSubFormField",
                "type": "string"
              },
              {
                "key": "prefix2.formWithSubFormKey.subForm",
                "type": "string"
              },
              {
                "lastKey": "lastValue"
              }
            ]
          }
